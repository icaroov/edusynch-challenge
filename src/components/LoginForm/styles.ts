import styled, { css } from 'styled-components'

export const Container = styled.div`
  ${({ theme }) => css`
    display: flex;
    flex-direction: column;

    max-width: 525px;

    background-color: ${theme.colors.primary};

    button {
      align-self: flex-end;
    }
  `}
`

export const Header = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: flex-start;

  letter-spacing: -0.06em;

  margin-bottom: 4.8rem;
`

export const Title = styled.h1`
  ${({ theme }) => css`
    color: ${theme.colors.white};

    font-size: 3.4rem;
    line-height: 118.5%;
    font-family: ${theme.font.secondary};
    font-weight: ${theme.font.light};
  `}
`

export const Subtitle = styled.h2`
  ${({ theme }) => css`
    color: ${theme.colors.tertiary};

    font-size: 3.4rem;
    line-height: 118.5%;
    font-weight: ${theme.font.bold};
    text-transform: uppercase;
  `}
`

export const Form = styled.form`
  ${({ theme }) => css`
    display: flex;
    flex-direction: column;

    input {
      margin-bottom: ${theme.spacings.small};
    }
  `}
`

export const CloseButton = styled.button`
  ${({ theme }) => css`
    border: none;
    font-size: 0;
    transition: 0.3s;
    background-color: transparent;

    img {
      width: 2rem;
      cursor: pointer;
    }

    &:focus {
      outline-width: 1px;
      outline-style: dashed;
      outline-color: ${theme.colors.mainBg};
    }
    &:hover {
      opacity: 0.7;
    }
  `}
`
