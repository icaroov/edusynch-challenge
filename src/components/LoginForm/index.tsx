import { useState } from 'react'

import Button from 'components/Button'
import Input from 'components/Input'

import * as Styled from './styles'

const LoginForm = () => {
  const [username, setUsername] = useState('')
  const [password, setPassword] = useState('')

  return (
    <Styled.Container>
      <Styled.Header>
        <Styled.Title>Get Started</Styled.Title>
        <Styled.Subtitle>Just Login</Styled.Subtitle>
      </Styled.Header>

      <Styled.Form>
        <Input
          label="Username"
          name="Username"
          value={username}
          setValue={setUsername}
        />

        <Input
          label="Password"
          name="Password"
          type="password"
          value={password}
          setValue={setPassword}
        />
      </Styled.Form>

      <Button>Login</Button>
    </Styled.Container>
  )
}

export default LoginForm
