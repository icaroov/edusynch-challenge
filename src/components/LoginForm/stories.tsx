import { Story, Meta } from '@storybook/react/types-6-0'

import LoginForm from '.'

export default {
  title: 'LoginForm',
  component: LoginForm
} as Meta

export const Default: Story = () => (
  <div style={{ maxWidth: '525px' }}>
    <LoginForm />
  </div>
)
