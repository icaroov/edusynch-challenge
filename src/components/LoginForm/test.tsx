import { screen } from '@testing-library/react'

import { renderWithTheme } from 'utils/tests/helpers'

import LoginForm from '.'

describe('<LoginForm />', () => {
  it('should render the header form', () => {
    renderWithTheme(<LoginForm />)

    expect(
      screen.getByRole('heading', { name: /get started/i })
    ).toBeInTheDocument()

    expect(
      screen.getByRole('heading', { name: /just login/i })
    ).toBeInTheDocument()
  })

  it('should render the form correctly', () => {
    const { container } = renderWithTheme(<LoginForm />)

    expect(screen.getByLabelText(/username/i)).toBeInTheDocument()

    expect(screen.getByLabelText(/password/i)).toBeInTheDocument()

    expect(screen.getByRole('button', { name: /login/i })).toBeInTheDocument()

    expect(container.parentElement).toMatchSnapshot()
  })
})
