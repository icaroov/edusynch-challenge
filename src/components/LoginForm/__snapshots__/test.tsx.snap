// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoginForm /> should render the form correctly 1`] = `
.c7 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  border: 0;
  padding: 1.5rem 4.4rem;
  border-radius: 0.4rem;
  font-weight: 700;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #7A57FD;
  background: #FFD74F;
  -webkit-transition: all 0.2s ease;
  transition: all 0.2s ease;
  cursor: pointer;
  height: 4.8rem;
  font-size: 1.4rem;
}

.c7:hover {
  background: #ffc70b;
}

.c5 {
  font-size: 1.6rem;
  color: #FFFFFF;
  margin-bottom: 0.5rem;
}

.c6 {
  width: 100%;
  max-height: 5.6rem;
  padding: 2rem 1.5rem;
  border-radius: 0.4rem;
  border: 1px solid #ccc;
  color: #FFFFFF;
  background-color: #9a7ffd;
  font-size: 1.8rem;
}

.c6:focus {
  outline-width: 1px;
  outline-style: dashed;
  outline-color: #6135D7;
}

.c6::-webkit-input-placeholder {
  color: #FFFFFF;
  font-size: 1.8rem;
}

.c6::-moz-placeholder {
  color: #FFFFFF;
  font-size: 1.8rem;
}

.c6:-ms-input-placeholder {
  color: #FFFFFF;
  font-size: 1.8rem;
}

.c6::placeholder {
  color: #FFFFFF;
  font-size: 1.8rem;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  background-color: #7A57FD;
}

.c0 button {
  -webkit-align-self: flex-end;
  -ms-flex-item-align: end;
  align-self: flex-end;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-letter-spacing: -0.06em;
  -moz-letter-spacing: -0.06em;
  -ms-letter-spacing: -0.06em;
  letter-spacing: -0.06em;
  margin-bottom: 4.8rem;
}

.c2 {
  color: #FFFFFF;
  font-size: 3.4rem;
  line-height: 118.5%;
  font-family: Poppins,-apple-system,BlinkMacSystemFont,'Segoe UI',Roboto,Oxygen,Ubuntu,Cantarell,'Open Sans','Helvetica Neue',sans-serif;
  font-weight: 300;
}

.c3 {
  color: #FFEAA4;
  font-size: 3.4rem;
  line-height: 118.5%;
  font-weight: 700;
  text-transform: uppercase;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c4 input {
  margin-bottom: 2.4rem;
}

@media screen and (max-width:768px) {
  .c0 button {
    width: 100%;
  }
}

<body>
  <div>
    <div
      class="c0"
    >
      <div
        class="c1"
      >
        <h1
          class="c2"
        >
          Get Started
        </h1>
        <h2
          class="c3"
        >
          Just Login
        </h2>
      </div>
      <form
        class="c4"
      >
        <div
          class=""
        >
          <label
            class="c5"
            for="Username"
          >
            Username
          </label>
          <input
            class="c6"
            id="Username"
            type="text"
            value=""
          />
        </div>
        <div
          class=""
        >
          <label
            class="c5"
            for="Password"
          >
            Password
          </label>
          <input
            class="c6"
            id="Password"
            type="password"
            value=""
          />
        </div>
      </form>
      <button
        class="c7"
      >
        <span>
          Login
        </span>
      </button>
    </div>
  </div>
</body>
`;
